src/logic/__tests__/saveProjectTest.js

```diff
@@ -0,0 +1,44 @@
+import saveProject from '../saveProject';
+import reduce from '../../reducers';
+import {userAuthenticated} from '../../actions/user';
+import {projectCreated, updateProjectSource} from '../../actions/projects';
+
+import {
+  credentialFactory,
+  userFactory,
+} from '@factories/clients/firebase';
+
+const defaultState = reduce(undefined, {type: 'NONE'});
+
+jest.mock('../../clients/firebase.js');
+
+test('not pristine project should save', async() => {
+  const mockCredential = credentialFactory.build();
+  const mockUser = userFactory.build();
+  let state = reduce(defaultState, projectCreated('123456'));
+  state = reduce(state, updateProjectSource('123456', 'css', ''));
+  state = reduce(state, userAuthenticated(mockUser, [mockCredential]));
```

Comment:
No big deal, but we can do this a little more concisely by using an actual `reduce`:

```js
const state = [
  projectCreated('123456'),
  updateProjectSource('123456', 'css', ''),
  userAuthenticated(mockUser, [mockCredential]),
].reduce(reduce, defaultState);